File: Frontend/src/classes/SportStaff.ts
import SportActivity from "./SportActivity";
import User from "./User";
import SportManager from './SportManager'


export default class SportStaff extends User{

    //Constructor
    public constructor(name:string, mail:string, password:string, role:string, address:string, phoneNum:string, hesCode:string){
        super(name, mail, password, role, address, phoneNum, hesCode);
    }

    //Methods
    public searchIndividual(ID:number):SportActivity | null{
        // TO DO
        return null;
    }
    public enterAbsenceStatus(ownerUID: string, OID:string){
        let sportManager = SportManager.getInstance();
        sportManager.changeAbsenceStatus(ownerUID, OID);
    }
}

File: Backend/src/main/java/com/example/backend/SportStaff.java
package com.example.backend;

import lombok.Getter;
import lombok.Setter;

@Setter
@Getter
public class SportStaff extends User{

    //Constructor
    public SportStaff(String name, String mail, String password, String role, String address, String phoneNum, String hesCode){
        super(name, mail, password, role, address, phoneNum, hesCode);
    }

    //Methods
    public SportActivity searchIndividual(int ID){
        // TO DO
        return null;
    }
    public boolean enterAbsenceStatus(SportActivity sportActivity){
        // TO DO
        return false;
    }
}
